src/fdb5/database/EntryVisitMechanism.h

```diff
@@ -52,7 +51,7 @@ class EntryVisitor : public eckit::NonCopyable {
 
 private: // methods
 
-    virtual void visitDatum(const Field& field, const Key& key) = 0;
+    virtual void visitDatum(const Field& field, const TypedKey& datumKey) = 0;
 
 protected:  // members
 
```

Comment:
not in this PR but:
good practice would be to init those members by {nullptr}

anyways, in source cc, `currentStore_` is not even initialized. should be fixed